tests/Constraints/CoerciveTest.php

```diff
@@ -11,271 +11,201 @@
 
 use JsonSchema\Constraints\Constraint;
 use JsonSchema\Constraints\Factory;
-use JsonSchema\SchemaStorage;
-use JsonSchema\Uri\UriResolver;
+use JsonSchema\Constraints\TypeCheck\LooseTypeCheck;
 use JsonSchema\Validator;
 
-class CoerciveTest extends BasicTypesTest
+class CoerciveTest extends VeryBaseTestCase
 {
-    /**
-     * @dataProvider getValidCoerceTests
-     */
-    public function testValidCoerceCasesUsingAssoc($input, $schema)
-    {
-        $checkMode = Constraint::CHECK_MODE_TYPE_CAST | Constraint::CHECK_MODE_COERCE_TYPES;
-
-        $schemaStorage = new SchemaStorage($this->getUriRetrieverMock(json_decode($schema)));
-        $schema = $schemaStorage->getSchema('http://www.my-domain.com/schema.json');
-
-        $validator = new Validator(new Factory($schemaStorage, null, $checkMode));
+    protected $factory = null;
 
-        $value = json_decode($input, true);
-
-        $validator->validate($value, $schema, $checkMode);
-        $this->assertTrue($validator->isValid(), print_r($validator->getErrors(), true));
+    public function setUp()
+    {
+        $this->factory = new Factory();
+        $this->factory->setConfig(Constraint::CHECK_MODE_TYPE_CAST | Constraint::CHECK_MODE_COERCE_TYPES);
     }
 
-    /**
-     * @dataProvider getValidCoerceTests
-     */
-    public function testValidCoerceCases($input, $schema, $errors = array())
+    public function dataCoerceCases()
     {
-        $checkMode = Constraint::CHECK_MODE_TYPE_CAST | Constraint::CHECK_MODE_COERCE_TYPES;
-
-        $schemaStorage = new SchemaStorage($this->getUriRetrieverMock(json_decode($schema)));
-        $schema = $schemaStorage->getSchema('http://www.my-domain.com/schema.json');
-
-        $validator = new Validator(new Factory($schemaStorage, null, $checkMode));
-        $value = json_decode($input);
-
-        $this->assertTrue(gettype($value->number) == 'string');
-        $this->assertTrue(gettype($value->integer) == 'string');
-        $this->assertTrue(gettype($value->boolean) == 'string');
+        // check type conversions
+        $types = array(
+            // toType
+            'string' => array(
+                //    fromType      fromValue       toValue         valid   Test Number
+                array('string',     '"string"',     'string',       true),  // #0
+                array('integer',    '45',           '45',           true),  // #1
+                array('boolean',    'true',         'true',         true),  // #2
+                array('boolean',    'false',        'false',        true),  // #3
+                array('NULL',       'null',         '',             true),  // #4
```

Comment:
Would you prefer them as separate methods? I like them better all in one place like this, because it's easier to compare them - but I can certainly break to one method per case if you'd rather.